pyqg/kernel.pyx

```diff
@@ -55,6 +55,15 @@ cdef class PseudoSpectralKernel:
     cdef DTYPE_com_t [:, :, :] dqhdt_p
     cdef DTYPE_com_t [:, :, :] dqhdt_pp
 
+    # subgrid velocity parameterizations
+    cdef DTYPE_real_t [:, :, :] du
+    cdef DTYPE_real_t [:, :, :] dv
+    cdef DTYPE_com_t [:, :, :] duh
+    cdef DTYPE_com_t [:, :, :] dvh
+    # subgrid potential vorticity parameterizations
+    cdef DTYPE_real_t [:, :, :] dq
+    cdef DTYPE_com_t [:, :, :] dqh
```

Comment:
Since there is no state about the parameterization that has to be persisted, I am curious why we need to add all of this stuff to the base model class.

Maybe the goal is to be able to include this stuff in diagnostics?

Reply:
`dqh`, `duh`, and `dvh` are being used in diagnostics [here](https://github.com/pyqg/pyqg/blob/7d50c93fdfc23997cfed2d691b102344f3b7f1ff/pyqg/model.py#L623-L665). We could omit their real counterparts, however, if we also follow your suggestion below to use function-local variables. I'm not certain about whether that would be less efficient, though (e.g. it seems like there might be performance benefits to initializing the FFT plans and memory for the arrays once at the top, rather than reinitializing / reallocating everything every timestep).